package.json

```diff
@@ -20,8 +20,9 @@
     "test-check-coverage": "npm run test-coverage && nyc check-coverage",
     "test-cloud": "mochify --wd --no-detect-globals --timeout=10000",
     "test-coverage": "nyc --all --reporter text --reporter html --reporter lcovonly npm run test-node",
-    "test": "npm run lint && npm run test-node && npm run test-headless",
-    "bundle": "browserify --no-detect-globals -s FakeTimers -o fake-timers.js src/fake-timers-src.js",
+    "test-esm": "node -r esm test-esm/fake-timers-esm-test.js",
```

Comment:
can we test this using actual node esm instead of the `esm` package? It's unflagged on node 13 and 14, and will probably be unflagged in next minor of node 12 (https://github.com/nodejs/node/pull/33055).

Just skip the tests on nodes that don't support it

Reply:
its not the same. node's ESM implementation is still vastly different to the browser's ESM implementation because of all the interop cruft it has and limitations.

the esm module just happens to be a half way house.

really we should be testing it in node and the browser directly, to make sure both their module systems work fine with it. but its a task worthy of its own pr i think

Reply:
Node's implementation is at least spec compliant, using `esm` is just a hack with broken semantics.

I don't disagree it should be tested in real browsers, but if testing in Node it should be testing with a real implementation, not a some weird FrankenEsm. E.g. `esm` module still allows `require`, named import of CJS, omitting file extension, directory import etc which are all against spec. Having a test passing with `esm` is no proof ESM is implemented correctly.

It's stricter if you use an `.mjs` extension<sup>*</sup>, but since we don't in this case that doesn't affect things.

FWIW, loading the generated file works fine with Node's built in ESM (as long as it's named `.mjs`)

<sup>*</sup>) almost compliant, expect it allows directory imports and leaving out the extension (maybe more, didn't test)